### PR TITLE
Add pushToGithub Feature

### PR DESCRIPTION
**Pull request by alisalehi1380**
Hello
Thank you for this repository. It is great and practical and I appreciate you.🙏

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
In my opinion, saving table data on the browser has limitations.
Therefore, I decided to fix this problem for myself. With my very little knowledge, I added a feature that uploads the SQL code to GitHub when I press `cntl + alt + s`.

**Additional context**
photo_2024-09-08_16-22-54

https://github.com/user-attachments/assets/615fbc51-ef86-4e69-a731-5e52232fbccc

My purpose in opening the PR was to show how I did it.
You definitely have much better handwriting than I do. 💪

### CODE REVIEW

File: src/utils/exportSQL/sentToGithub.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import axios from 'axios';
+
+const TOKEN = 'YOUR_GITHUB_TOKEN';
```
Comment (1ilit) [inline src/utils/exportSQL/sentToGithub.js:3]: This is unacceptable. Tokens do not go in source files. Also this won't work in production

File: src/utils/exportSQL/sentToGithub.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    const headers = {
+        'Authorization': `token ${TOKEN}`,
+        'Accept': 'application/vnd.github.v3+json',
+        'Content-Type': 'application/json',
+    };
+
+    try {
+        // Step 1: Create a Blob
+        let response = await axios.post(`https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/git/blobs`, {
```
Comment (1ilit) [inline src/utils/exportSQL/sentToGithub.js:18]: drawdb uses octokit to interface with the github api

File: src/utils/exportSQL/sentToGithub.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            sha: commitSha,
+            force: true,
+        }, {headers});
+
+        alert('Commit created successfully. ✅');
+    } catch (error) {
+        alert('Error creating commit. ❌');
+    }
+
+}
```
Comment (1ilit) [inline src/utils/exportSQL/sentToGithub.js:62]: Not linted or formatted

File: src/utils/exportSQL/sentToGithub.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        }, {headers});
+        const commitSha = response.data.sha;
+
+        // Step 5: Update the reference
+        await axios.patch(`https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/git/refs/heads/master`, {
+            sha: commitSha,
+            force: true,
+        }, {headers});
+
+        alert('Commit created successfully. ✅');
```
Comment (1ilit) [inline src/utils/exportSQL/sentToGithub.js:57]: You can do better than alerts

**Review by 1ilit — COMMENTED** [review]
Sorry the implementation is lacking in many ways. Is this ai generated?

**Comment by alisalehi1380** [comment]
Yes, as I said, I know very little JavaScript and React.
I got help from chat GPT.

**Comment by 1ilit** [comment]
Thanks for the contribution but chatgpt isn't reliable and sucks even more when it comes to larger code bases

**Pull request closed without merging**